core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -39,27 +42,117 @@
+import org.elasticsearch.search.suggest.term.TermSuggestionBuilderTests;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 
 import java.io.IOException;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map.Entry;
 
+import static org.hamcrest.Matchers.containsString;
+
 public class SuggestBuilderTests extends WritableTestCase<SuggestBuilder> {
 
+    private static NamedWriteableRegistry namedWriteableRegistry;
+    private static IndicesQueriesRegistry queriesRegistry;
+    private static ParseFieldMatcher parseFieldMatcher;
+    private static Suggesters suggesters;
+
+    /**
+     * Setup for the whole base test class.
+     */
+    @BeforeClass
+    public static void init() {
+        NamedWriteableRegistry nwRegistry = new NamedWriteableRegistry();
+        nwRegistry.registerPrototype(SuggestionBuilder.class, TermSuggestionBuilder.PROTOTYPE);
+        nwRegistry.registerPrototype(SuggestionBuilder.class, PhraseSuggestionBuilder.PROTOTYPE);
+        nwRegistry.registerPrototype(SuggestionBuilder.class, CompletionSuggestionBuilder.PROTOTYPE);
+        nwRegistry.registerPrototype(SmoothingModel.class, Laplace.PROTOTYPE);
+        nwRegistry.registerPrototype(SmoothingModel.class, LinearInterpolation.PROTOTYPE);
+        nwRegistry.registerPrototype(SmoothingModel.class, StupidBackoff.PROTOTYPE);
+        namedWriteableRegistry = nwRegistry;
+        queriesRegistry = new SearchModule(Settings.EMPTY, namedWriteableRegistry).buildQueryParserRegistry();
+        suggesters = new Suggesters(new HashMap<>());
+        parseFieldMatcher = ParseFieldMatcher.STRICT;
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        namedWriteableRegistry = null;
+        queriesRegistry = null;
+        suggesters = null;
+        parseFieldMatcher = null;
+    }
 
     @Override
     protected NamedWriteableRegistry provideNamedWritableRegistry() {
-        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry();
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, TermSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, PhraseSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, CompletionSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, Laplace.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, LinearInterpolation.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, StupidBackoff.PROTOTYPE);
         return namedWriteableRegistry;
     }
 
     /**
+     * Test that valid JSON suggestion request passes.
+     */
+    public void testValidJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
```

Comment:
I'd say we cover most of parsing syntactically correct xContent already in the `testFromXContent`, except for randomizing the order in which the fields are parsed which is fixed when we use our builders.toXContent. Is this test adding anything on top of that? If no, I think we don't need it.